test/core/libp2p/crypto/key_generator_test.cpp

```diff
@@ -0,0 +1,448 @@
+      "LXhxxFuoJZvjvgdgtiDbQsQ+XzTWHsycrqDgLT5EiAd+em1sDYIMuAv4h7XfigQc\n"
+      "G3H5gtpoarZ30s2WztBZ8A35g5KrewmYCXhQpAIFtc+qtSezCtk9XEJtyvO3+Ddd\n"
+      "EHM2uviOB0pdxrjjSr02bKFhVzEpP0xFD8T5PKJSaAwx79/ur3UVipfEDc0TdzbV\n"
+      "OzYDLZ0gZQFPARvnAb5dPf102+nHJO2xWf4adss7Um8ayHlYrWFFRap4GGc6iCEb\n"
+      "BQ3GXTPKv5DxyZjN7BwF/6oIzq8YNjlNVxr+3X2gTSfePF1vY2R0FrkL2XIMXxm6\n"
+      "fnCQgJPXox4Sa9NejO74K0+K/JvNu7rAM/x8gfKxHyKsQ6vkMASNR/wEqD4TJxa5\n"
+      "CspA92XQHWOxifEINwoTUJRJCR4yMGlmuP8rWi8hKZZVQtVlb1VS/A/HlIkK5fKk\n"
+      "4g1x1lLPtc4V4Y6nO2+M7t1QOpYnyEjcq7uMpOaain7X8/61Qw/9Xf+HYGkdRt80\n"
+      "Sg6XnyaBS5nw5FkVrzk0nK8aRRGPjcuY2YJU48gGximJ64dS14nflAy0rxP7BOWt\n"
+      "B+QG9IIdxBXuYRXEgFLj789iXcOEbRDhggqxSEJlC/tK94/TJ2sY3r26sTNswCzw\n"
+      "jAvoo38wj4uht4tsOJzFAwmCTi0XslA1S+8kluWRapaUCK1GAHGS5OAespZFjgmM\n"
+      "FfJtzxSv7fbfP+cxFmyGEcHH4BNqG6eLIAvF/LnPjlzLrOdDaXBRyLQYVsYTHxhv\n"
+      "YyDAo85g4wA9l9FlrxzSeUCIOycS/MSlr80N9f30vKlQsdvCXQVsJujIBpxeLeBy\n"
+      "f8VAKF4ovytEdIPuEeAdxN/vgDndFShYEKzFzWvJ1j13kH6nFLNSlSy8gD6ipQ2h\n"
+      "ueMnOSif5Vu22finZo6AhQI0GRUm/DJH4VJh1wHi3uwlAqHVRgsUBq4mBR2jt/Ar\n"
+      "w8fTejx3BQJ8YYwTysNy9gJ3qyNVjPuHMcwInn/HtbMzEQTH56XPXDac1ywAA9F4\n"
+      "/ibA8wFjSlM41xl1HRXe9OKkWnX9SmeiJReMkxLKm4wXA1R1771yWtrHt0Eme47W\n"
+      "/ZmpSiDPGQSHESj8s/Mn955s4tMrFfHZpsb4KiQ5mhevDhSpJ4MjNpUQImI6KjVA\n"
+      "6LKwDtG4msFWmWb7ObLCZo1G3mL494/3T52Nmc+meBz3hPeuD8VGeHq92b1sISQQ\n"
+      "7HYbhvO08Niu8pPDpJsNv06HNE80s6HrjhoXC+9goMef8Y0NjUF5YGb0luEpbvb9\n"
+      "LcU8O2XRmUeoGOwbawTDoEHFXBRfdyTAmjpH9G2KFVK/wI8Avn+kiE2E5D2wREYk\n"
+      "ekmMyFngICZtuECzPjkIMJrLQ3GdcpUKu3i9sfGP6hVWcUc0V4pJC9GnKG5keChd\n"
+      "8z8vEv5Nsh6zeDfEkf/D//GLVUyaBIFaEMGLWgVdX08/mHBLfuDgzPpEnEukZeMt\n"
+      "fApKUpUEqaveDgHPybxdMRj8bHayKLP+bPdRsxrDaeudEu8V5m11zRO6q4iTuqXk\n"
+      "RocDmNl7WEZAxgRDnmJgTakvJgmZOfql4u5cmRcJOl6hYn+8f7452ig8QUgyvziW\n"
+      "tAKZvHmm6Wi4bUVwW2NNRK55WrpS1q5EkCQXAAZ3NJZ6cEl3KeBVh+qRHrUosgb4\n"
+      "AM1NVw6Dv2GJR3mpKwGRQTW119VrzmzUnqd8WXkkTYmR1Z7xjAidJCdyDKrWUgGB\n"
+      "bZ57PxTo5nV/wX9Sb82LUcKrYEyh4Uf3hoiU+AGXzDeAoH5KckRCk6DQHtiAD8yr\n"
+      "pPO6mYhqadKCT0n9049n3KP+aLEUHbi6GOhrm/ahUNcx8toQ/tB9QHK4NPIefhvx\n"
+      "nZNYElZHS1s5rYtDHiALsGz8jRn8J38/M/c9cf2T3IrrqPgacKsXJIVFreHekEf1\n"
+      "uJ6WaZMYbdIrH6tuBYYfiIqsuoL6w8A5P8xE51wc1RiY/FVlMZfLwdJyYGd9xay9\n"
+      "-----END RSA PRIVATE KEY-----";
+
+  std::string ed25519_private_key_ =
+      "-----BEGIN PRIVATE KEY-----\n"
+      "MC4CAQAwBQYDK2VwBCIEIBdcqTY4Ybi+EeJEOzX/GwsN5IYZp4pndoES2tb473I6\n"
+      "-----END PRIVATE KEY-----";
+
+  std::string secp256k1_private_key_ =
+      "-----BEGIN EC PRIVATE KEY-----\n"
+      "MHQCAQEEIDmfPdQEwSJiWRP/8nDQjUtRfVtLMp+9VddWbPB8d1UcoAcGBSuBBAAK\n"
+      "oUQDQgAEflsDwKkQagCUyoo+luEyOD+07Evx6Sg4FkfmMoWNr5/ux2sKpC6379TN\n"
+      "+R8zS1oIhsnnUZHJBUE5TN8yC108kQ==\n"
+      "-----END EC PRIVATE KEY-----";
+};
+
+class KeyTest : public virtual testing::Test {
+ protected:
+  class TempFile {
```

Comment:
Too complicated approach IMO
Instead we can create a files during SetUp and remove during TearDown. Take a look at BaseFsTest example

Reply:
refactored